wordpress-mobile/WordPress-iOS
# Show switch to block editor in empty classic post

This brings WPiOS in-line with WPAndroid where users in the classic editor can switch to the block editor as long as the post is empty and the site defaults to Gutenberg.

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/16293

To test:
1. Locate a site that has Gutenberg enabled
2. Locate a post created using the classic editor (you might need to temporarily switch the site over to classic by turning the "Enable block editor" switch off in Site Settings, creating a classic post, and switching the site back to Gutenberg)
3. Open the post for editing and remove all content (the title can be left as-is)
4. Tap the three-dots menu and expect to see a "Switch to block editor" option

## Regression Notes
1. Potential unintended areas of impact

The changes to the app's logic only affect the rendering of the "Switch to block editor" action in the classic editor menu bottom sheet. Barring a crash, I don't think these changes could affect other parts of the app.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I smoke tested the change and added a new test.

3. What automated tests I added (or what prevented me from doing so)

I added the `AztecPostViewController+MenuTests` unit tests to help catch regressions if the behavior implemented here breaks.

PR submission checklist:
